tools/roslaunch/src/roslaunch/xmlloader.py

```diff
@@ -294,8 +299,19 @@ def _arg_tag(self, tag, context, ros_config, verbose=True):
         try:
             self._check_attrs(tag, context, ros_config, XmlLoader.ARG_ATTRS)
             (name,) = self.reqd_attrs(tag, context, ('name',))
-            value, default, doc = self.opt_attrs(tag, context, ('value', 'default', 'doc'))
             
+            params_restore_point = copy.deepcopy(context.params)
+            resolve_dict_restore_point = copy.deepcopy(context.resolve_dict)
+            # Use context to emulate get_param functionality in substitution_args
+            for key,v in ros_config.params.items():
+                p = Param(key,v)
```

Comment:
``ros_condfig.params`` is usually a map from ``str`` to  ``Param``, so you should be able to add `v` directly.

Reply:
I'll look it in the weekend. Thanks for your comments.